netdiff/utils.py

```diff
@@ -134,35 +179,26 @@ def _netjson_networkgraph(
         raise NetJsonError('version cannot be None except when protocol is "static"')
     # prepare nodes
     node_list = []
-    for node in nodes:
-        netjson_node = OrderedDict({'id': node[0]})
+    for ip, data in nodes:
+        netjson_node = OrderedDict({'id': ip})
         # must copy properties dict to avoid modifying data
-        properties = node[1].copy()
-        local_addresses = properties.pop('local_addresses', None)
-        label = properties.pop('label', None)
-        # append local_addresses only if not empty
-        if local_addresses:
-            netjson_node['local_addresses'] = local_addresses
-        # append properties only if not empty
-        if properties:
-            netjson_node['properties'] = properties
-        # append label only if not empty
-        if label:
-            netjson_node['label'] = label
+        properties = data.copy()
+        netjson_node['label'] = properties.pop('label', '') or ''
+        netjson_node['local_addresses'] = properties.pop('local_addresses', [])
+        netjson_node['properties'] = properties or {}
         node_list.append(netjson_node)
+    node_list.sort(key=lambda d: d['id'])
     # prepare links
     link_list = []
-    for link in links:
+    for src, dst, props in links:
```

Comment:
I prefer if we keep using non abbreviated variable names, `source`, `dest`, `properties`